nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js

```diff
@@ -0,0 +1,2298 @@
+                    if (isChecked) {
+                        parameter.value = '';
+                    } else {
+                        parameter.value = null;
+                    }
+                }
+
+                // add a row for the new parameter
+                parameterData.addItem(parameter);
+
+                // sort the data
+                parameterData.reSort();
+
+                // select the new parameter row
+                var row = parameterData.getRowById(parameterCount);
+                parameterGrid.setActiveCell(row, parameterGrid.getColumnIndex('value'));
+                parameterCount++;
+            } else {
+                nfDialog.showOkDialog({
+                    headerText: 'Parameter Exists',
+                    dialogContent: 'A parameter with this name already exists.'
+                });
+
+                // select the existing properties row
+                var matchingRow = parameterData.getRowById(matchingParameter.id);
+                parameterGrid.setSelectedRows([matchingRow]);
+                parameterGrid.scrollRowIntoView(matchingRow);
+            }
+
+            // close the new parameter dialog
+            $('#parameter-dialog').modal('hide');
+
+            // update the buttons to possibly trigger the disabled state
+            $('#parameter-context-dialog').modal('refreshButtons');
+
+        } else if (!parameterKeyRegex.test(parameterName)) {
+            nfDialog.showOkDialog({
+                headerText: 'Configuration Error',
+                dialogContent: 'This parameter appears to have an invalid character or characters. Only alpha-numeric characters (a-z, A-Z, 0-9), hyphens (-), underscores (_), periods (.), and spaces ( ) are accepted.'
+            });
+        } else {
+            nfDialog.showOkDialog({
+                headerText: 'Configuration Error',
+                dialogContent: 'The name of the parameter must be specified.'
+            });
+        }
+    };
+
+    /**
+     * Update a parameter.
+     */
+    var updateParameter = function () {
+        var parameterName = $.trim($('#parameter-name').val());
+
+        // ensure the parameter name is specified
+        if (parameterName !== '') {
+            var parameterGrid = $('#parameter-table').data('gridInstance');
+            var parameterData = parameterGrid.getData();
+
+            // ensure the parameter name is unique
+            var matchingParameter = null;
+            $.each(parameterData.getItems(), function (_, item) {
+                if (parameterName === item.name) {
+                    matchingParameter = item;
+                    return false;
+                }
+            });
+
+            if (matchingParameter !== null) {
+                var isChecked = $('#parameter-dialog').find('.nf-checkbox').hasClass('checkbox-checked');
+
+                var parameter = {
+                    id: matchingParameter.id,
+                    hidden: false,
```

Comment:
need to also set
```affectedComponents: matchingParameter.affectedComponents,```

otherwise, the update blows away the affected components and when you select another param, then back to the edited one, the referenced components panel is blank.